### PR TITLE
Closes #23 ADVS template program

### LINKED ISSUE #23: Template: ADVS
Showcase from start to finish how to create a full obesity ADVS. ​This should include:
- Parameter mappings​
- Endpoint programming​
- Any new computations​
Start work once #22 is merged.

Use [{admiral} ADVS template](https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_advs.R) as a starting point.

### PR DESCRIPTION
**Pull request by AndersAskeland**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by AndersAskeland** [comment]
Think it should be ready for review now. Feel free to review @pharmaverse/admiralmetabolic .

I've used a somewhat different layout/setup compared to the other admiral packages. For me it makes more sense, but please let me know if it appears strange.

File: inst/ad_advs.R [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+# Assign parameter variables ----
+# See the "Assign PARAMCD, PARAM, PARAMN, PARCAT1" vignette section for more
+# information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#paramcd)
+
+# Add all parameter variables (PARAM, PARAMN, PARCAT1)
+advs <- advs |>
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD, PARAM, PARAMN, PARCAT1, PARCAT1N),
```
Comment (manciniedoardo) [inline inst/ad_advs.R:227]: Is `PARAM` required here as it's already in ADVS?

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks @AndersAskeland!

I did a quick review and left some initial comments - I haven't tired executing the template yet. 

In general, I am wondering if we would benefit from building on the existing ADVS template from {admiral} (rather than starting afresh like here) and just adding our extra bits? It would make review easier as well. interested in your thoughts.

File: inst/ad_advs.R [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+
+# Define look-up tables ----
+# See function documentation for `derive_vars_merged_lookup()` and
+# `derive_vars_cat()` for information of how look-up tables can be used:
+# (https://pharmaverse.github.io/admiral/reference/derive_vars_merged_lookup.html)
+# (https://pharmaverse.github.io/admiral/reference/derive_vars_cat.html)
+
+# Parameter look-up
+param_lookup <- tribble(
```
Comment (starosto) [inline inst/ad_advs.R:23]: we agreed that for PARAMN in range 1-5 PARCAT1 = "Anthropometric measurements"

File: inst/ad_advs.R [hunk 3]
```diff
@@ -0,0 +1,260 @@
+# Name: ADVS
```
Comment (starosto) [inline inst/ad_advs.R:1]: should be in templates folder /inst/templates/ad_advs.R

File: inst/ad_advs.R [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+# See the "Derive Additional Parameters" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#derive_param)
+
+# Derive BMI
+advs <- advs |>
+  filter(VSTESTCD != "BMI") |>
+  derive_param_bmi(
+    by_vars = exprs(
+      STUDYID, USUBJID, TRTSDT, TRTEDT, TRT01P, TRT01A, VISIT,
+      VISITNUM, ADT, ADY, VSTPT, VSTPTNUM
```
Comment (starosto) [inline inst/ad_advs.R:137]:
```suggestion
      STUDYID, USUBJID, TRTSDT, TRTEDT, TRT01P, TRT01A, AVISIT,
      AVISITN, ADT, ADY, ATPT, ATPTN
```

File: inst/ad_advs.R [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+    ),
+    set_values_to = exprs(
+      PARAMCD = "BMI",
+      DOMAIN = "VS"
```
Comment (starosto) [inline inst/ad_advs.R:141]:
```suggestion
      PARAMCD = "BMI"
```

File: inst/ad_advs.R [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#baseline)
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#bchange)
+
+# Add baseline flag (ABLFL)
+advs <- advs |>
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+      order = exprs(ADT, VSTPTNUM, VISITNUM),
```
Comment (starosto) [inline inst/ad_advs.R:174]:
```suggestion
      order = exprs(ADT, ATPTN, AVISITN),
```

File: inst/ad_advs.R [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+  derive_var_pchg()
+
+
+# Derive criterion variables ----
+# See the "Derive Criterion Variables" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#crit_vars)
+
+# Set weight loss criterion flags (CRIT1, CRIT1FL)
+advs <- advs |>
+  derive_vars_crit_flag(
```
[the remark this answers is not in the document]
Comment (starosto) [reply inst/ad_advs.R:202]: or filter = (AVISITN > 0 & PARAMCD == "WEIGHT")

**Review by starosto — COMMENTED** [review]
Hi @AndersAskeland , I have posted some comments. Well done!

File: inst/ad_advs.R [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+  "BMI",    AVAL >= 35 & AVAL < 40,    "Obesity class II",  5,
+  "BMI",    AVAL >= 40,                "Obesity class III", 6,
+  "BMI",    is.na(AVAL),               NA_character_,       NA_integer_
+)
+
+# Read in data ----
+# See the "Read in Data" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#readdata)
+
+# Read data
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R:52]: After thinking more about it, I agree :)

File: inst/ad_advs.R [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+vs_metabolic <- admiralmetabolic::vs_metabolic
+adsl <- admiral::admiral_adsl
+
+# Convert SAS empty values to NA
+advs <- vs_metabolic |>
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R:57]: Yeah, I have actually been thinking about this a lot and going back and forth.

I agree that we need to use the same pipe functionality across the package. I am however personally of the opinion that using the native pipe provides several benefits, among other that we don't need a {magrittr} dependency and I believe it will be more future proof.

However, I also agree that we should try to align to Admiral and the other extension packages. Hence, if Admiral have no plan of ever changing over to the native pipe, I agree that I should change to {magrittr}.

File: inst/ad_advs.R [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+  convert_blanks_to_na()
+adsl <- adsl |>
+  convert_blanks_to_na()
+
+# Merge ADSL variables (TRTSDT, TRTEDT, TRT01P, TRT01A) needed for ADVS
+# derivations
+advs <- advs |>
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = exprs(TRTSDT, TRTEDT, TRT01P, TRT01A),
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R:67]: I actually disagree here. I personally believe that a template like this is often used nonsequential, where having arguments stored into objects can be confusing. 

Furthermore, not having it stored in an object enables us to not use the injection operator (`!!!`) later in the code, which is in my opinion a fairly advanced feature which will confuse people.

File: inst/ad_advs.R [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+    by_vars = exprs(STUDYID, USUBJID)
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R:68]: I actually strongly disagree here. If we assume that people will use these templates as something to base their own work on, extracting argument data from an package specific metadata system is a bad idea. 

I can see that this makes a lot of sense inside R code, but think that we should strongly consider if this is a good idea for a template.

File: inst/ad_advs.R [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+
+# Attach/load required packages ----
+library(admiral)
+library(admiralmetabolic)
+library(tibble)
+library(dplyr)
+library(stringr)
+
+
+# Define look-up tables ----
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R:16]: What do you mean? Would you place the library calls directly before using any functionalities within that package? I.e. that we load stringr() right before the code where we actually use it?

File: inst/ad_advs.R (hunk 2 above)
Comment (AndersAskeland) [reply inst/ad_advs.R:23]: Yeah, I could not remember what we agreed. We should also update the vignette.

File: inst/ad_advs.R (hunk 12 above)
[the remark this answers is not in the document]
Comment (starosto) [reply inst/ad_advs.R:16]: I would place "https://pharmaverse.github.io/admiral/reference/derive_vars_merged_lookup.html" link directly before using the function derive_vars_merged_lookup(), same with derive_vars_cat()

File: inst/ad_advs.R (hunk 9 above)
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply inst/ad_advs.R:57]: I think best for now would be going with magrittr, can revisit later.

File: inst/ad_advs.R (hunk 11 above)
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply inst/ad_advs.R:68]: @AndersAskeland I see your point. It's a tough thing to balance because we _have_ these options and so we should:
1.  let the users be aware they exist, 
2.  use them ourselves in templates.

In general I agree with your concerns though- in fact within admiralroche we have the following code at the top of our Roche-specific templates:
```
# Set subject keys ----
# user to set subject key variables in one place

set_admiral_options(subject_keys = exprs(STUDYID, USUBJID))
```
We could do the same in our templates?

File: inst/ad_advs.R (hunk 9 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R:57]: Will change for now. But I will also create an issue about potentially changing it :)

File: inst/ad_advs.R (hunk 7 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R:202]: Agree

File: inst/ad_advs.R (hunk 1 above)
Comment (AndersAskeland) [reply inst/ad_advs.R:227]: It is not added during the early join (only PARAMCD is joined) as that is the way the Admiral vignette does it.

File: inst/ad_advs.R [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+      STUDYID, USUBJID, TRTSDT, TRTEDT, TRT01P, TRT01A, VISIT,
+      VISITNUM, ADT, ADY, VSTPT, VSTPTNUM
+    ),
+    set_values_to = exprs(
+      PARAMCD = "BMI",
+      DOMAIN = "VS"
+    ),
+    get_unit_expr = VSSTRESU,
+    filter = VSSTAT != "NOT DONE" | is.na(VSSTAT),
+    constant_by_vars = exprs(USUBJID)
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R:145]: Which part? I am unsure about the filter, but it is based on our current vignette. Do you know why we have used a filter @Siddhesh2097 ?

File: inst/ad_advs.R (hunk 12 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R:16]: The lookup table creations were moved a bit around, where they now reside under their respective use places. That actually means that the functions are not referenced directly. Instead the respective BDS vignette part is referenced, which should have some description about lookup tables. I think this is adequate.

File: inst/ad_advs.R (hunk 2 above)
Comment (AndersAskeland) [reply inst/ad_advs.R:23]: It is changed. I will update the issue for the vignette before resolving the conversation.

File: inst/ad_advs.R (hunk 4 above)
Comment (AndersAskeland) [reply inst/ad_advs.R:137]: Nice catch!

File: inst/ad_advs.R (hunk 7 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R]: We should actually also remember to update this in the ADVS vignette.

File: inst/ad_advs.R (hunk 1 above)
Comment (manciniedoardo) [reply inst/ad_advs.R]: You're right sorry, I must've mistaken this block for the one at the beginning where only `PARAMCD` is merged.

File: inst/ad_advs.R (hunk 10 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R]: I am still very hesitant about this change. I will implement it now, but I think it needs to be revisited in the future. I personally believe as Admiral becomes more widespread, we would need to lower the bar of entry. Especially considering many people coming in are coming from SAS with little to no experience in R.

Would also love the team (@pharmaverse/admiralmetabolic) to also give their input.

File: inst/ad_advs.R (hunk 13 above)
[the remark this answers is not in the document]
Comment (Siddhesh2097) [reply inst/ad_advs.R]: This is as per the [BDS finding vignette](https://pharmaverse.github.io/admiral/articles/bds_finding.html#derive_param). But it's not required in this case as per the vs_metabolic data.
I think we should remove it as it creates unnecessary confusion to the user.
@AndersAskeland  @manciniedoardo

[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R]: Thanks for checking @Siddhesh2097 ! I will remove it from the template then.

**Comment by AndersAskeland** [comment]
@pharmaverse/admiralmetabolic If anyone have time, please have a review of the template.

File: inst/templates/ad_advs.R [hunk 14]
```diff
@@ -0,0 +1,269 @@
...
+# Assign parameter variables ----
+# See the "Assign PARAMCD, PARAM, PARAMN, PARCAT1" vignette section for more
+# information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#paramcd)
+
+# Add all parameter variables (PARAM, PARAMN, PARCAT1, PARCAT1N)
+advs <- advs %>%
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD, PARAM, PARAMN, PARCAT1, PARCAT1N),
```
Comment (starosto) [inline inst/templates/ad_advs.R:235]:
```suggestion
    dataset_add = select(param_lookup, -VSTESTCD),
```

File: inst/ad_advs.R (hunk 12 above)
[the remark this answers is not in the document]
Comment (starosto) [reply inst/ad_advs.R]: @AndersAskeland thanks for the update.
Please consider to add  link to the function/BDS vignette part right before derive_vars_merged_lookup() function call.

File: inst/templates/ad_advs.R [hunk 15]
```diff
@@ -0,0 +1,269 @@
...
+library(admiralmetabolic)
+library(tibble)
+library(dplyr)
+library(stringr)
+
+
+# Define project options/variables ----
+# Use the admiral option functionality to store subject key variables in one
+# place
+set_admiral_options(subject_keys = exprs(STUDYID, USUBJID))
```
Comment (starosto) [inline inst/templates/ad_advs.R:19]:
```suggestion
```
subject_keys defaults to exprs(STUDYID, USUBJID), we don't need to set this up

**Review by starosto — COMMENTED** [review]
@AndersAskeland I added some comments. I also wonder if we should not build on existing ADVS templates from {admiral}. If not now, then for future templates.

**Review by manciniedoardo — APPROVED** [review]
LGTM @AndersAskeland 😄 (after comments from @starosto are addressd)

File: inst/ad_advs.R (hunk 12 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/ad_advs.R]: I have moved it, and have a link to the BDS vignette part around line 230. Are you looking in the newest version?

File: inst/templates/ad_advs.R [hunk 16]
```diff
@@ -0,0 +1,269 @@
...
+adsl <- admiral::admiral_adsl
+
+# Convert SAS empty values to NA
+advs <- vs_metabolic %>%
+  convert_blanks_to_na()
+adsl <- adsl %>%
+  convert_blanks_to_na()
+
+# Merge ADSL variables (stored in `adsl_vars`) needed for ADVS
+# derivations
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:41]: I disagree on this, as the main reason we are adding the ADSL variables are for later derivations, and having a "subsection" about deviations could confuse. I have however changed it to "... needed for ADVS" to avoid confusion. Do you agree?

File: inst/templates/ad_advs.R [hunk 17]
```diff
@@ -0,0 +1,271 @@
...
+# Derive Date/Time and Analysis Day ----
+# See the "Derive/Impute Numeric Date/Time and Analysis Day" vignette section
+# for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#datetime)
+
+# Add vital sign analysis date (ADT) and treatment start date (TRTSDT)
+advs <- advs %>%
+  derive_vars_dt(new_vars_prefix = "A", dtc = VSDTC)
+advs <- advs %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
```
Comment (yurovska) [inline inst/templates/ad_advs.R:85]:
```suggestion
advs <- advs %>%
  derive_vars_dt(new_vars_prefix = "A", dtc = VSDTC) %>%
  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
```

File: inst/templates/ad_advs.R [hunk 18]
```diff
@@ -0,0 +1,271 @@
...
+    by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD)),
+    source_var = AVAL,
+    new_var = BASE
+  )
+
+# Derive absolute (CHG) and relative (PCHG) change from baseline
+advs <- advs %>%
+  derive_var_chg()
+advs <- advs %>%
+  derive_var_pchg()
```
Comment (yurovska) [inline inst/templates/ad_advs.R:195]:
```suggestion
advs <- advs %>%
  derive_var_chg() %>%
  derive_var_pchg()
```

**Review by yurovska — CHANGES_REQUESTED** [review]

File: inst/templates/ad_advs.R [hunk 19]
```diff
@@ -0,0 +1,271 @@
...
+# simplified usage throughout the program
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01P, TRT01A)
+
+
+# Read in data ----
+# See the "Read in Data" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#readdata)
+
+# Read data
+vs_metabolic <- admiralmetabolic::vs_metabolic
```
Comment (yurovska) [inline inst/templates/ad_advs.R:31]:
```suggestion
vs_metabolic <- admiralmetabolic::vs_metabolic %>%
  filter(
    !(VSSTAT %in% "NOT DONE")
  )
  
```

**Review by yurovska — CHANGES_REQUESTED** [review]

File: inst/templates/ad_advs.R [hunk 20]
```diff
@@ -0,0 +1,271 @@
...
+    AVISIT = case_when(
+      is.na(VISIT) ~ NA_character_,
+      str_detect(VISIT, "SCREEN|UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      TRUE ~ str_to_title(VISIT)
+    ),
+    AVISITN = case_when(
+      VISIT == "BASELINE" ~ 0,
+      str_detect(VISIT, "WEEK") ~ as.integer(str_extract(VISIT, "\\d+")),
+      TRUE ~ NA_integer_
+    )
```
Comment (yurovska) [inline inst/templates/ad_advs.R:106]: Needs to be revised as `AVISIT/AVISITN` are always blank for `HEIGHT` (measured at `SCREENING 1` only).

File: inst/templates/ad_advs.R [hunk 21]
```diff
@@ -0,0 +1,271 @@
...
+# See the "Derive Additional Parameters" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#derive_param)
+
+# Derive BMI
+advs <- advs %>%
+  filter(VSTESTCD != "BMI") %>%
+  derive_param_bmi(
+    by_vars = exprs(
+      STUDYID, USUBJID, !!!adsl_vars,
+      AVISITN, ADT, ADY, ATPT, ATPTN
```
Comment (yurovska) [inline inst/templates/ad_advs.R:129]:
```suggestion
      AVISIT, AVISITN, ADT, ADY, ATPT, ATPTN
```

`AVISIT` never gets populated if not in `by_vars`.

**Review by yurovska — CHANGES_REQUESTED** [review]

File: inst/templates/ad_advs.R [hunk 22]
```diff
@@ -0,0 +1,271 @@
...
+    new_vars = adsl_vars,
+    by_vars = get_admiral_option("subject_keys")
+  )
+
+# Define parameter look-up table used for merging parameter codes to ADVS
+param_lookup <- tribble(
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN, ~PARCAT1, ~PARCAT1N,
+  "HEIGHT", "HEIGHT", "Height (cm)", 1, "Anthropometric measurement", 1,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 2, "Anthropometric measurement", 1,
+  "BMI", "BMI", "Body Mass Index(kg/m^2)", 3, "Anthropometric measurement", 1,
```
Comment (yurovska) [inline inst/templates/ad_advs.R:53]:
```suggestion
  "BMI", "BMI", "Body Mass Index(kg/m2)", 3, "Anthropometric measurement", 1,
```

image

**Review by yurovska — CHANGES_REQUESTED** [review]

File: inst/templates/ad_advs.R (hunk 19 above)
Comment (Siddhesh2097) [reply inst/templates/ad_advs.R:31]: @yurovska There is no VSSTAT == "NOT DONE" in the current vs data and thus this filter was removed from both vignette and template to avoid confusion to the user as to why this filter was added.

Comment (yurovska) [reply inst/templates/ad_advs.R:31]: I know there are no cases like that but thought it would be nice to do that just in case the vs data changes in the future. I'm not insisting though.

File: inst/templates/ad_advs.R (hunk 16 above)
[the remark this answers is not in the document]
Comment (starosto) [reply inst/templates/ad_advs.R]: @AndersAskeland I followed the template admiral/inst/templates/ad_advs.R, but if you disagree, I am fine with the current order

File: inst/templates/ad_advs.R (hunk 15 above)
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:19]: The reason why its manually set, is too highlight that we will be using this functionality through the template. I feel that this should be kept, but have added some text explaining that the default is STUDYID and USUBJID.

File: inst/templates/ad_advs.R (hunk 18 above)
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:195]: Agree

File: inst/templates/ad_advs.R (hunk 15 above)
Comment (manciniedoardo) [reply inst/templates/ad_advs.R:19]: I like this as well @AndersAskeland, no problem if we deviate a bit from ADVS in admiral core.

File: inst/templates/ad_advs.R (hunk 19 above)
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:31]: We actually removed this earlier and also removed it from the vignette. But I think we might want to revisit this later, as it could make sense to have this (both here and in the vignette), when its in our dataset. We should perhaps consider adding some records to the data as well.

Comment (yurovska) [reply inst/templates/ad_advs.R:31]: No objections to keeping it out for now and revising later.

File: inst/templates/ad_advs.R (hunk 20 above)
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:106]: Good catch! I don't want to deviate to much from the standard admiral BDS template, so my suggestion would be to have a added check for height. 
```
advs %>%
    mutate(
        ATPT = VSTPT,
        ATPTN = VSTPTNUM,
        AVISIT = case_when(
            is.na(VISIT) ~ NA_character_,
            VSTESTCD == "HEIGHT" & VISIT == "SCREENING 1" ~ "Screening",
            str_detect(VISIT, "SCREEN|UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
            TRUE ~ str_to_title(VISIT)
        ),
        AVISITN = case_when(
            VISIT == "BASELINE" ~ 0,
            VSTESTCD == "HEIGHT" & VISIT == "SCREENING 1" ~ -1,
            str_detect(VISIT, "WEEK") ~ as.integer(str_extract(VISIT, "\\d+")),
            TRUE ~ NA_integer_
        )
    ) 
```

Comment (yurovska) [reply inst/templates/ad_advs.R]: To be honest, I thought it would be "Baseline" as well, but "Screening" is fine too.

File: inst/templates/ad_advs.R [hunk 23]
```diff
@@ -0,0 +1,277 @@
...
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#baseline)
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#bchange)
+
+# Add baseline flag (ABLFL)
+advs <- advs %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(!!!get_admiral_option("subject_keys"), PARAMCD),
+      order = exprs(ADT, ATPTN, AVISITN),
```
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:177]: This dataset is a bit strange when it comes to blood pressure. In reality, what one usually does in blood pressure is that you measure the blood pressure a minimum of three times, and report the average of the last two measurements. It is from my perspective unclear what we actually have in the test data.

However, this strange grouping actually happened after we changed from using "VISITNUM" to "AVISITN" in this function.

I have just quickly looked at the admiral BDS template, where it appears that they actually have many baseline values and have change for each basetype. I am changing back to VISITNUM, and then we can later discuss if this might be confusing.

File: inst/templates/ad_advs.R (hunk 20 above)
Comment (AndersAskeland) [reply inst/templates/ad_advs.R]: I don't know precisely how AVISIT should be defined, but the ADAMIG states "AVISIT should be unique for a given analysis visit window.". While not entirely clear, I am interpreting this as, "The AVISIT cannot be the same for visits defined as different visit windows such as screening and baseline."

Comment (kathrinflunkert) [reply inst/templates/ad_advs.R]: In my opinion it is definitely CDISC compliant to have VISIT == 'SCREENING' and VISIT == 'BASELINE' on the SDTM side, but define a combined **analysis** visit window of AVISIT == 'BASELINE' on ADaM.
As you cited: "AVISIT should be unique for a given **analysis** visit window." But the **analysis** visit window can be different than the window on the collection side.

File: inst/templates/ad_advs.R (hunk 23 above)
[the remark this answers is not in the document]
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:177]: I spoke to soon. It appears that this is not the case. The different ATPT's are actually used as basetypes in the admiral vignette (https://pharmaverse.github.io/admiral/articles/bds_finding.html#baseline). 

I don't think adding basetypes to our vignette would make to much sense, and would argue that the way it is (where we use the last measurements at baseline) might be best for now. I do still agree that it is messy, but cannot think of anything that makes perfect sense right now.

File: inst/templates/ad_advs.R (hunk 20 above)
Comment (AndersAskeland) [reply inst/templates/ad_advs.R]: Thanks for your input @kathrinflunkert ! Reading it again now, I agree with you! I'll go ahead and change it to baseline :)

**Comment by AndersAskeland** [comment]
Thanks for all the comments and suggestions. Most are now implemented - if you want, have a quick look. It would be great to have it merged before our meeting tomorrow.

File: inst/templates/ad_advs.R (hunk 23 above)
[the remark this answers is not in the document]
Comment (yurovska) [reply inst/templates/ad_advs.R:177]: Ok, let's keep that in mind and potentially change it in one of the next PRs.

**Comment by AndersAskeland** [comment]
> I don't see a commit changing `AVISIT` from `Screening` to `Baseline` for `HEIGHT` as discussed [here](https://github.com/pharmaverse/admiralmetabolic/pull/35#discussion_r1829470709). Otherwise, LGTM! Thanks for working on that!

Apparently I got an fatal error on my push attempt - Is up now. Thanks for your comments and review @yurovska .

File: inst/templates/ad_advs.R [hunk 24]
```diff
@@ -0,0 +1,276 @@
...
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html)
+
+# Derive analysis time point (ATPT, ATPTN) and analysis visit (AVISIT, AVISITN)
+advs <- advs %>%
+  mutate(
+    ATPT = VSTPT,
+    ATPTN = VSTPTNUM,
+    AVISIT = case_when(
+      is.na(VISIT) ~ NA_character_,
+      VSTESTCD == "HEIGHT" & VISIT == "SCREENING 1" ~ "BASELINE",
```
Comment (yurovska) [inline inst/templates/ad_advs.R:98]:
```suggestion
      VSTESTCD == "HEIGHT" & VISIT == "SCREENING 1" ~ "Baseline",
```

**Review by yurovska — CHANGES_REQUESTED** [review]

File: inst/templates/ad_advs.R (hunk 24 above)
Comment (AndersAskeland) [reply inst/templates/ad_advs.R:98]: Woops! Thanks!

**Review by yurovska — APPROVED** [review]

**Comment by manciniedoardo** [comment]
@kathrinflunkert @Siddhesh2097 @starosto I will be merging in a few hours. If you have any final notes, please share now, otherwise please approve!

File: inst/templates/ad_advs.R (hunk 23 above)
[the remark this answers is not in the document]
Comment (starosto) [reply inst/templates/ad_advs.R:176]:
```suggestion
    derivation = derive_var_extreme_flag,
    args = params(
      by_vars = exprs(!!!get_admiral_option("subject_keys"), PARAMCD, ATPTN),
      order = exprs(ADT, ATPTN, AVISITN),
```
IMO we should flag with ABLFL=Y not only records with ATPT = "AFTER STANDING FOR 3 MINUTES". 
Why not add ATPTN to the by_vars function argument?

I also agree with @AndersAskeland that the average of SYSBP/DIABP measurements is often taken into account. This means that another parameter should be created. Please decide if it would be beneficial to the user.

**Review by starosto — APPROVED** [review]
@AndersAskeland besides the comment on the ABLFL, I have no further thoughts. Thanks for a great job 💯
